lib/exabgp/bgp/message/update/attribute/bgpls/linkstate.py

```diff
@@ -59,8 +59,10 @@ def unpack(cls, data, negotiated):
 			klass.TLV = scode
 			ls_attrs.append(klass)
 			data = data[length+4:]
+		# Remove all but last instance of each TLV class - the last holds state for all TLVs of that type
+		ls_attrs = list({ls_attr.TLV: ls_attr for ls_attr in ls_attrs}.values())
 		for klass in ls_attrs:
-			if hasattr(klass, 'terids'):
+			if hasattr(klass, 'terids') or hasattr(klass, 'sr_adj_sids') or hasattr(klass, 'sr_adj_lan_sids'):
```

Comment:
```suggestion
			if hasattr(klass, 'reset'):
```
This might be cleaner